modin/pandas/test/test_groupby.py

```diff
@@ -1790,42 +1792,50 @@ def test_not_str_by(by, as_index):
     "agg_func",
     [
         pytest.param(
-            lambda grp: grp.apply(lambda df: df.dtypes), id="modin_dtype_impl"
+            lambda grp: grp.apply(lambda df: df.dtypes), id="modin_dtypes_impl"
         ),
-        pytest.param(lambda grp: grp.apply(lambda df: df.sum()), id="apply(sum)"),
+        pytest.param(lambda grp: grp.apply(lambda df: df.sum()), id="apply_sum"),
         pytest.param(lambda grp: grp.count(), id="count"),
         pytest.param(lambda grp: grp.nunique(), id="nunique"),
+        # Integer key means the index of the column to replace it with.
+        # 0 and -1 are considered to be the indices of the columns to group on.
+        pytest.param({1: "sum", 2: "nunique"}, id="dict_agg_no_intersection_with_by"),
         pytest.param(
-            {"c": "sum", "d": "nunique"}, id="dict_agg_no_intersection_with_by"
```

Comment:
why did you remove usage of named columns here?

Reply:
I did this because we don't know the exact column names of the testing frame here, so I just used their numerical indices to compute the actual labels when the info about the frame will be available.